service/controller/resource/machinepoolupgrade/resource.go

```diff
@@ -84,9 +91,13 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			return microerror.Mask(err)
 		}
 	}
-
 	r.logger.LogCtx(ctx, "level", "debug", "message", "ensured release labels are set on respective AzureMachinePool")
 
+	err = r.ensureLastDeployedReleaseVersion(ctx, &cr)
```

Comment:
Updating `release.giantswarm.io/last-deployed-version` annotation for the `MachinePool` CR.